observation_portal/common/state_changes.py

```diff
@@ -92,28 +91,28 @@ def on_request_state_change(old_request, new_request):
                 modify_ipp_time_from_requests(ipp_value, [new_request], 'credit')
 
 
-@transaction.atomic
-def on_requestgroup_state_change(old_requestgroup, new_requestgroup):
-    if old_requestgroup.state == new_requestgroup.state:
+def on_requestgroup_state_change(old_requestgroup_state, new_requestgroup):
+    if old_requestgroup_state == new_requestgroup.state:
         return
-    valid_request_state_change(old_requestgroup.state, new_requestgroup.state, old_requestgroup)
-    if new_requestgroup.state in TERMINAL_REQUEST_STATES:
-        for r in new_requestgroup.requests.filter(state__in=['PENDING']):
-            r.state = new_requestgroup.state
-            r.save()
+    valid_request_state_change(old_requestgroup_state, new_requestgroup.state, new_requestgroup)
+    if new_requestgroup.state in ['CANCELED', 'WINDOW_EXPIRED']:
```

Comment:
changed this because I think it should only update the Request state if the RG was not COMPLETED. We don't want PENDING requests set to COMPLETED...

Reply:
Yeah, good call